src/main/java/me/rayzr522/jsonmessage/ReflectionHelper.java

```diff
@@ -0,0 +1,61 @@
+package me.rayzr522.jsonmessage;
+
+import org.bukkit.Bukkit;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.Optional;
+
+public class ReflectionHelper {
+
+    private static final String version;
+    private static final int MAJOR_VER;
+
+    static {
+        String[] split = Bukkit.getServer().getClass().getPackage().getName().split("\\.");
+        version = split[split.length - 1];
+        MAJOR_VER = Integer.parseInt(version.split("_")[1]);
+    }
+
+    public static Class<?> getClass(String path) throws ClassNotFoundException {
+        return Class.forName(path.replace("{nms}", "net.minecraft.server." + version).replace("{obc}", "org.bukkit.craftbukkit." + version));
+    }
+
+    public static Optional<Field> findFieldByTypeName(Class<?> targetClass, String fieldTypeName) {
+        return Arrays.stream(targetClass.getDeclaredFields())
+                .filter(field -> field.getType().getSimpleName().equals(fieldTypeName))
+                .findFirst();
+    }
+
+    public static Field getFieldByTypeName(Class<?> targetClass, String fieldTypeName) {
+        return findFieldByTypeName(targetClass, fieldTypeName)
+                .orElseThrow(() -> new IllegalArgumentException(String.format(
+                        "Could not find field with type '%s' in class %s",
+                        fieldTypeName, targetClass.getCanonicalName()
+                )));
+    }
+
+    public static void setFieldValue(Field field, Object instance, Object value) {
+        try {
+            field.set(instance, value);
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
```

Comment:
yea, error handling as a whole is something i'd like to improve later. but it can have a pretty drastic impact on downstream consumers and this has been the behavior so far, so for now I'm going to leave it as is